src/Command/FillContributionsStatsCommand.php

```diff
@@ -36,7 +62,8 @@ public function execute(Arguments $args, ConsoleIo $io)
 
         // Fetch
         $contributions = $this->Contributions->find()
-                              ->where(["type !=" => "license", "action !=" => "update"])
+                              ->where(["type !=" => "license", "action !=" => "update",
+                                       "datetime >=" => $from, "datetime <=" => $to])
```

Comment:
The parameter `$to` won’t work like you thought.

Mysql stores the datetime as `YYYY-MM-DD HH:MM:SS`. Let’s say `$to = '2020-01-01'`. If there is a record with `datetime = 2020-01-01 01:23:45`, then it won’t match because the string comparison `'2020-01-01 01:23:45' <= '2020-01-01'` is false. Actually, it can never be equal because you are comparing a date with a datetime.

If I was to write a test for this, it could be:
```php
    function testExecute_paramsGiven_ContributionsAreRewrittenToTo() {
        $this->exec("fill_contributions_stats -f 2016-06-19 -t 2016-06-19");

        $firstContributionStats = $this->ContributionsStats->find()->order(['id' => 'DESC'])->first();
        $this->assertEquals('2016-06-19', $firstContributionStats->date);
    }
```

Reply:
MySQL knows how to deal with comparison between a datetime and a string. However, I did make a silly mistake because I forgot that the string will be interpreted as the datetime with 00:00:00. I just need to replace that with $lastDay + 1 day, and it should be okay.

I will update a fix later.